oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -101,6 +101,21 @@ public static GoogleCredentials getApplicationDefault() throws IOException {
     return getApplicationDefault(OAuth2Utils.HTTP_TRANSPORT_FACTORY);
   }
 
+  /**
+   * <<<<<<< HEAD
+   * Returns the gcloud CLI's well-known location path. Corresponds to one of the elements of {@code
+   * =======
+   * Returns the gcloud CLI's well-known location file. Corresponds to one of the elements of {@code
+   * >>>>>>> 608179607cce89b7d8c3271ffd0d5a078a457255
+   * getApplicationDefault()}, exposed for convenience.
+   *
+   * @return the credentials, or null if not found
+   * @throws IOException if the credentials cannot be created in the current environment.
+   */
+  public static String getWellKnownCredentialsPath() {
+    return defaultCredentialsProvider.getWellKnownCredentialsPath();
```

Comment:
This is sketchy. Because here we rely on the fact that default credentials provider is static and always the same. This could change and if we decide to change it - this method will makes things more complicated. 

How about you move the logic of the getWellKnownCredentialsPath into a new public  Utils class, like GoogleAuthUtils. (all the existing utils classes are internal and therefore private) We already have some other public static methods that could be moved to the new Utils class as well.

Reply:
I created a `GoogleAuthUtils` with a `getPath` method that would use the default provider instance to resolve the path. It also has internal methods to use a test provider